### PR TITLE
Add inspect_callstack_entry

### PR DESCRIPTION
**Pull request by wyozi**
For some stack inspection action

### CODE REVIEW

File: ducc/src/ducc.rs [hunk 1]
```diff
@@ -274,6 +274,53 @@ impl Ducc {
...
+    /// 
+    /// ```
+    /// # use ducc::Ducc;
+    /// # let ducc = Ducc::new();
+    /// if let Some(entry) = ducc.inspect_callstack_entry(-2) {
+    ///    let file_name = entry.function.into_object().get::<_, String>("fileName").unwrap();
+    ///    println!("Function was called from {} line {}", file_name, entry.line_number);
+    /// };
+    /// ```
+    pub fn inspect_callstack_entry<'ducc>(&'ducc self, level: i32) -> Option<CallStackEntry<'ducc>> {
```
Comment (wyozi) [inline ducc/src/ducc.rs:296]: This could be a `Result`, but would it be worth it for a new `ErrorKind` just for calling the function with invalid level?

Comment (SkylerLipthay) [reply ducc/src/ducc.rs:296]: I think this API is good as-is! I would just add a note to the doc comment that `None` is returned if the given level goes beyond the current callstack depth.

**Review by SkylerLipthay — CHANGES_REQUESTED** [review]
I like this! If you could write a formal test I'll merge ASAP.

File: ducc/src/tests/ducc.rs [hunk 2]
```diff
@@ -31,6 +31,16 @@ fn no_duktape_global() {
...
+#[test]
+fn inspect_callstack() {
+    let ducc = Ducc::new();
+    ducc.globals().set("fun", ducc.create_function(|inv| {
+        inv.ducc.inspect_callstack_entry(-1).expect("callstack entry was None");
+        Ok(())
+    })).unwrap();
+    ducc.globals().get::<_, Function>("fun").unwrap().call::<(), ()>(()).unwrap();
+}
+
```
Comment (SkylerLipthay) [inline ducc/src/tests/ducc.rs:48]: Just add `use function::Function;` to this file and I'll merge! (it doesn't compile currently)

Comment (wyozi) [reply ducc/src/tests/ducc.rs:48]: Juggling with Github editor has its downsides :d

Comment (SkylerLipthay) [reply ducc/src/tests/ducc.rs:48]: Indeed... with your latest change you don't need to `use function::Function` at all ;)

**Comment by SkylerLipthay** [comment]
🎉

**Comment by SkylerLipthay** [comment]
Oh by the way I changed from `inspect_callstack_entry` to `inspect_call_stack_entry`... Duktape internally switched between the two styles and I figured we should be consistent and choose just one.

**Pull request merged**